### PR TITLE
Elo voting per user shuffle

### PR DESCRIPTION
**Pull request by aidanmorgan**
We now use a per-user cache of session id's (shuffled) to draw from rather than a shared pool, this will ensure that users will see every talk continually.

Uses a cookie set on the client browser to create a 24 hour session id, with the collection of session ids stored in CosmosDB (with a TTL set).

### CODE REVIEW

File: DDD.Functions/EloVotingGetPair.cs [hunk 1]
```diff
@@ -19,44 +19,26 @@ namespace DDD.Functions
 {
     public static class EloVotingGetPair
     {
+        private const string CookieName = "DDDPerth.VotingSessionId";
```
Comment (mzaatar) [inline DDD.Functions/EloVotingGetPair.cs]: Does it worth have a different flag for test/prod env.
Something like isProd? "dddperth.voteSessionId" : "test.dddperth.voteSessionId"

File: DDD.Functions/local.settings.json [hunk 2]
```diff
@@ -5,31 +5,37 @@
...
     "AzureWebJobsDashboard": "UseDevelopmentStorage=true",
     "FUNCTIONS_WORKER_RUNTIME": "dotnet",
+
     // Conference
     "ConferenceInstance": "2019",
+
     // SessionizeReadModelSync
     "SessionizeReadModelSyncSchedule": "* * * * *",
-    "SessionsConnectionString": "UseDevelopmentStorage=true",
+    "SessionsConnectionString": "DefaultEndpointsProtocol=https;AccountName=dddperthdatatest;AccountKey=gxUQ5OeKRqmmS4qdz0w1xsPaA09fX+RrkzkCK4LAodq//kwJm0kj8Y9FGbyimtrVZahrQq7MmubncW+f68+FCg==;EndpointSuffix=core.windows.net",
```
Comment (mzaatar) [inline DDD.Functions/local.settings.json]: To remove.

File: DDD.Functions/local.settings.json [hunk 3]
```diff
@@ -51,19 +59,25 @@
     "WaitingListCanVoteWithEmail": "true",
     "MinVotes": "6",
     "MaxVotes": "12",
+
     // EloVoting
-    "EloVotesConnectionString": "UseDevelopmentStorage=true",
+    "EloVotesConnectionString": "DefaultEndpointsProtocol=https;AccountName=dddperthdatatest;AccountKey=gxUQ5OeKRqmmS4qdz0w1xsPaA09fX+RrkzkCK4LAodq//kwJm0kj8Y9FGbyimtrVZahrQq7MmubncW+f68+FCg==;EndpointSuffix=core.windows.net",
```
Comment (mzaatar) [inline DDD.Functions/local.settings.json]: Remove as well

File: DDD.Functions/local.settings.json [hunk 4]
```diff
@@ -51,19 +59,25 @@
...
     "EloVotingTable": "EloVotes",
     "EloPasswordPhrase":"Perth2022",
     "EloEnabled": "true",
     "EloAllowedTimeInSecondsToSubmit": "120",
+    "UserVotingSessionsConnectionString": "AccountEndpoint=https://dddperthdatatest.documents.azure.com:443/;AccountKey=UbR2GBvA5PPok2ctY7t5aXetTCDPokB9Z4BxJoeKj51JwfKsr97ql2022IUawL12QJAx9VJmM2nVq7jq94yeTg==;",
+    "UserVotingSessionsDatabaseId": "dddperthdatatest",
+    "UserVotingSessionsContainerId": "votesessions",
```
Comment (mzaatar) [inline DDD.Functions/local.settings.json:62]: Is it container or db table? A bit confusing for new people

**Review by mzaatar — CHANGES_REQUESTED** [review]
Reviewed it without running it locally. LGTM with required changes

File: DDD.Functions/EloVotingGetPair.cs (hunk 1 above)
Comment (aidanmorgan) [reply DDD.Functions/EloVotingGetPair.cs]: Good call - I'll make it configurable.

File: DDD.Functions/local.settings.json (hunk 4 above)
Comment (aidanmorgan) [reply DDD.Functions/local.settings.json:62]: These are the names that are used by CosmosDb, I agree they are confusing tho :-)

**Pull request merged**